src/runtime/template_helpers.ts

```diff
@@ -60,18 +60,24 @@ function withKey(elem: any, k: string) {
   return elem;
 }
 
-function prepareList(collection: any): [any[], any[], number, any[]] {
-  let keys: any[];
-  let values: any[];
+function prepareList(collection: unknown): [unknown[], unknown[], number, undefined[]] {
+  let keys: unknown[];
+  let values: unknown[];
 
-  if (Array.isArray(collection)) {
+  if (!collection || typeof collection !== "object") {
```

Comment:
you can do this check in the else clause, so the usual case is (very very slightly) faster